src/utils/index.js

```diff
@@ -177,29 +177,90 @@ const getOptionsData = (options) => {
 const parseCourseTypeOptions = (typeOptions) => {
   const courseRunTypeOptions = {};
   const runTypeModes = {};
-  const entitlementUUIDS = [];
-  const courseTypeOptions = [{ label: 'Select enrollment track', value: '' }].concat(typeOptions.map((type) => {
-    courseRunTypeOptions[type.uuid] = [{ label: 'Select enrollment track', value: '' }].concat(type.course_run_types.map((courseRunType) => {
+  const priceLabels = {};
+  const initialSelect = [{ label: 'Select enrollment track', value: '' }];
+  const courseTypeOptions = initialSelect.concat(typeOptions.map((type) => {
+    const runTypeOptions = type.course_run_types.map((courseRunType) => {
       runTypeModes[courseRunType.uuid] = courseRunType.modes;
       return { label: courseRunType.name, value: courseRunType.uuid };
-    }));
-    // This should be changed to remove the part about AUDIT_TRACK once we stop
-    // creating draft Audit entitlements. Right now, this is used to decide if we
-    // show the price on the course create/edit form.
-    if (type.entitlement_types.length && !(type.entitlement_types.length === 1 &&
-            type.entitlement_types[0] === AUDIT_TRACK.key)) {
-      entitlementUUIDS.push(type.uuid);
-    }
+    });
+    courseRunTypeOptions[type.uuid] = initialSelect.concat(runTypeOptions);
+
+    const typePrices = {};
+    type.tracks.forEach((track) => {
+      if (track.mode.payee !== '' && track.seat_type !== null) {
+        typePrices[track.seat_type.slug] = track.seat_type.name;
+      }
+    });
+    priceLabels[type.uuid] = typePrices;
+
     return { label: type.name, value: type.uuid };
   }));
   return {
-    entitlementUUIDS,
+    priceLabels,
     courseRunTypeOptions,
     courseTypeOptions,
     runTypeModes,
   };
 };
 
+const formatPriceData = (formData, courseOptions) => {
+  const priceData = {
+    price: 0, // DISCO-1399 - we can stop sending single price
+    prices: {},
+  };
+
+  if (!formData.type) { // DISCO-1399 can drop this format support
+    priceData.price = formData.price.replace(/\.00$/, '');
+    priceData.prices['professional'] = priceData.price;
+    priceData.prices['verified'] = priceData.price;
+    return priceData;
+  }
+
+  const courseOptionsData = getOptionsData(courseOptions);
+  if (!courseOptionsData || !formData.type) {
```

Comment:
Right you are. Dropped `!formData.type` - I'd rather keep the rest of the check as-is, because I want to keep it once DISCO-1399 cleanup happens. It's an early exit for the "with type" flow, which we'll still want.